lib/srv/discovery/discovery.go

```diff
@@ -40,6 +41,8 @@ import (
 	"github.com/gravitational/teleport/lib/srv/server"
 )
 
+var errNoInstances = trace.NotFound("all fetched nodes already enrolled")
```

Comment:
Declare `errNoInstancesMsg` as a string and use trace.NotFound where you actually returns the error. This saves a trace.Wrap call and captures a more correct stack trace.

```suggestion
const errNoInstancesMsg = "all fetched nodes already enrolled"
```

Reply:
@codingllama But then you need to use "string comparison" here https://github.com/gravitational/teleport/pull/21087/files#diff-7a030693b277c072225ebea353dbdf312158a3c04c4ee09da1befbef2180e7b3R472 or use `trace.IsNotFound()` and "hope" that this is the only "not found" error returned from that function. 
If we want to avoid the cost of `trace`, we can define this error with `errors.New()` and unwrap on a check.

Reply:
The `errors.New` suggestion is pretty solid, best of both worlds I think.

Is the unwrap necessary? I thought errors.Is would work with trace, but maybe I'm mistaken.

Reply:
trace works with errors.Is.